core/test/network-records-to-devtools-log.js

```diff
@@ -227,6 +244,48 @@ function addRedirectResponseIfNeeded(networkRecords, record) {
 function networkRecordsToDevtoolsLog(networkRecords, options = {}) {
   const devtoolsLog = [];
   networkRecords.forEach((networkRecord, index) => {
+    // If we're operating on network record raw objects (not NetworkRequest instances),
+    // then we're operating on test data that may need to be massaged a bit.
+    if (networkRecord.constructor === Object) {
+      // Temporary code while we transition away from startTime and endTime.
+      // This allows us to defer slightly changes to test files.
+      // TODO: remove after timing refactor is done
+      // See https://github.com/GoogleChrome/lighthouse/pull/14311
+      if (networkRecord.startTime !== undefined) {
+        networkRecord.networkRequestTime = networkRecord.startTime;
+        // Old tests never distinguished between these two.
+        networkRecord.rendererStartTime = networkRecord.startTime;
+      }
+      if (networkRecord.endTime !== undefined) {
+        networkRecord.networkEndTime = networkRecord.endTime;
+      }
+      if (networkRecord.responseReceivedTime !== undefined) {
+        networkRecord.responseHeadersEndTime = networkRecord.responseReceivedTime;
+      }
+
+      // Set timing.requestTime and timing.receiveHeadersEnd to be values that
+      // NetworkRequest will pull from for networkRequestTime and responseHeadersEndTime,
+      // so this roundtrips correctly. Unless, of course, timing values are explicitly set
+      // already.
+
+      // If networkRecord.timing explicitly does not having a timing object, only create one
+      // if we absolutely need it. See _recomputeTimesWithResourceTiming.
+      const netReqTime = networkRecord.networkRequestTime;
+      const willNeedTimingObject =
+        (netReqTime !== undefined && netReqTime !== networkRecord.rendererStartTime) ||
+        (networkRecord.responseHeadersEndTime !== undefined);
+      if (willNeedTimingObject) networkRecord.timing = networkRecord.timing || {};
```

Comment:
ugh, the problem here is that `networkRequestTime` isn't properly set when `timing` is missing, failing 8 tests in `network-request-test.js`.

For example, take `update fetch stats for Lightrider` test. the `startTime`/`networkRequestTime` value is now 50 instead of 0, because I wanted some separation between the first two timing properties. Without a timing object, such a distinction can never be made.

Should I 1) fix the tests to provide the relevant timing object to make the values come out right or 2) do a check in this records->log conversion if these properties differ, and only create a timing object magically...to reduce some test boilerplate. Leaning towards 1.

Reply:
maybe it would be easier to always create a timing object now?

Reply:
fwiw in the correct tests `willNeedTimingObject` is always false for uses-rel-preconnect ... and there are zero tests in that audit that exercise a missing timing object. Is this a valid scenario?

Reply:
bump for the above question

Reply:
> fwiw for current PR code, willNeedTimingObject is always false for uses-rel-preconnect ... and there are zero tests in that audit that exercise a missing timing object. Is this a valid scenario?

Maybe just for failed requests? I'm not sure if it's important, it's possible it's just inherited from super old SDK code, or it could handle some important case we're not thinking of :)